### PR TITLE
clarify workflow events

### PR DESCRIPTION
**Pull request by dbu**
i was a bit confused by what the events exactly mean. after reading through the code and testing some things, this is what i found.

### CODE REVIEW

File: workflow/usage.rst [hunk 1]
```diff
@@ -177,10 +177,15 @@ what actions are allowed on a blog post::
...
 object with an ``EventDispatcher``. You can now create event listeners to
-block transitions (i.e. depending on the data in the blog post). The following
-events are dispatched:
+block transitions (i.e. depending on the data in the blog post) and do
+additional actions when a workflow operation happened (i.e. sending 
+announcements). The following events are dispatched:
+
+* ``workflow.guard``
+* ``workflow.[workflow name].guard``
+* ``workflow.[workflow name].guard.[transition name]``
```
Comment (dbu) [inline workflow/usage.rst]: i found it illogic that these events are shown below, when they are also events, just a special type of events.

File: workflow/usage.rst [hunk 2]
```diff
@@ -177,10 +177,23 @@ what actions are allowed on a blog post::
...
+
+- An event for every workflow;
+- An event for the workflow concerned;
+- An event for the workflow concerned with the specific transition or place name;
+
+The following events are dispatched:
+
+* ``workflow.guard``
+* ``workflow.[workflow name].guard``
+* ``workflow.[workflow name].guard.[transition name]``
```
Comment (dbu) [inline workflow/usage.rst]: i found it illogic that these events are shown below, when they are also events, just a special type of events.

File: workflow/usage.rst [hunk 3]
```diff
@@ -198,6 +211,14 @@ events are dispatched:
 * ``workflow.[workflow name].announce``
 * ``workflow.[workflow name].announce.[transition name]``
 
+When a state transition is initiated, the events are fired in the following order:
+
+- guard: Validate whether the transition is allowed at all (see below);
```
Comment (weaverryan) [inline workflow/usage.rst]: Let's link the see below part

File: workflow/usage.rst [hunk 4]
```diff
@@ -198,6 +211,14 @@ events are dispatched:
...
+- leave: The object is about to leave a place;
+- transition: The object is going through this transition;
+- enter: The object entered a new place. This is the first event where the object' is marked as being in the new place;
+- announce: Triggered once for each workflow that now is available for the object.
```
Comment (weaverryan) [inline workflow/usage.rst]: I wonder if we could integrate this with the event list above so it's all in one place. Perhaps the name (guard), then indented below the description, then bullets below that with the 3 events. We do something here https://github.com/symfony/symfony-docs/blob/master/introduction/http_fundamentals.rst for GET POST stuff

**Review by weaverryan — APPROVED** [review]
Minor comments, but easy win!

**Comment by xabbuh** [comment]
Thank you David.

File: workflow/usage.rst (hunk 4 above)
Comment (dbu) [inline workflow/usage.rst]: thanks. see #8060 for this. it looks nice, at least in the github preview

Comment (xabbuh) [inline workflow/usage.rst]: Oh, I already opened #8059 in the meantime. 🙈

**Pull request merged**